entity-service/src/test/kotlin/com/egm/stellio/entity/service/EntityServiceTests.kt

```diff
@@ -1305,6 +1303,80 @@ class EntityServiceTests {
         confirmVerified()
     }
 
+    @Test
+    fun `it should delete all entity property instances`() {
+
+        val entityId = "urn:ngsi-ld:Beehive:123456"
+
+        every { neo4jRepository.hasPropertyOfName(any(), any()) } returns true
+        every { neo4jRepository.deleteEntityProperty(any(), any(), any()) } returns 1
+
+        entityService.deleteEntityAttributes(entityId, "fishNumber", aquacContext!!)
+
+        verify { neo4jRepository.hasPropertyOfName(match {
+            it.id == "urn:ngsi-ld:Beehive:123456" &&
+                it.label == "Entity"
+        }, "https://ontology.eglobalmark.com/aquac#fishNumber") }
+        verify { neo4jRepository.deleteEntityProperty(match {
+            it.id == "urn:ngsi-ld:Beehive:123456" &&
+                it.label == "Entity"
+        }, "https://ontology.eglobalmark.com/aquac#fishNumber", null, true) }
+
+        confirmVerified()
+    }
+
+    @Test
+    fun `it should delete an entity relationship instance with the provided datasetId`() {
+
+        val entityId = "urn:ngsi-ld:Beehive:123456"
+
+        every { neo4jRepository.hasPropertyOfName(any(), any()) } returns false
+        every { neo4jRepository.hasRelationshipInstance(any(), any(), any()) } returns true
+        every { neo4jRepository.deleteEntityRelationship(any(), any(), any(), any()) } returns 1
+
+        entityService.deleteEntityAttribute(entityId, "connectsTo", URI.create("urn:ngsi-ld:Dataset:connectsTo:01"), aquacContext!!)
+
+        verify { neo4jRepository.hasRelationshipInstance(match {
+            it.id == "urn:ngsi-ld:Beehive:123456" &&
+                it.label == "Entity"
+        }, any(), URI.create("urn:ngsi-ld:Dataset:connectsTo:01")) }
+
+        verify { neo4jRepository.deleteEntityRelationship(match {
+            it.id == "urn:ngsi-ld:Beehive:123456" &&
+                it.label == "Entity"
+        }, any(), URI.create("urn:ngsi-ld:Dataset:connectsTo:01"), false) }
+
+        confirmVerified()
+    }
+
+    @Test
+    fun `it should not delete all entity attribute instances if the attribute is not found`() {
+
+        val entityId = "urn:ngsi-ld:Beehive:123456"
+
+        every { neo4jRepository.hasPropertyOfName(any(), any()) } returns false
+        every { neo4jRepository.hasRelationshipOfType(any(), any()) } returns false
+
+        assertThrows<ResourceNotFoundException>("Attribute fishNumber not found in entity urn:ngsi-ld:Beehive:123456") {
```

Comment:
Beware that the message displayed here is not checked against the message raised by the exception, it is just an informative message in case the assert fails.

If you want to check the message raised by the method, you have to do something like:

```
val exception = assertThrows...
assertEquals("my expected message", exception.message)
```